TWLight/users/models.py

```diff
@@ -105,6 +105,112 @@ def create_user_profile(sender, instance, created, **kwargs):
 models.signals.post_save.connect(create_user_profile, sender=settings.AUTH_USER_MODEL)
 
 
+def editor_reg_date(identity, global_userinfo):
+    # Try oauth registration date first.  If it's not valid, try the global_userinfo date
+    try:
+        reg_date = datetime.strptime(identity["registered"], "%Y%m%d%H%M%S").date()
+    except (TypeError, ValueError):
+        try:
+            reg_date = datetime.strptime(
+                global_userinfo["registration"], "%Y-%m-%dT%H:%M:%SZ"
+            ).date()
+        except (TypeError, ValueError):
+            reg_date = None
+            pass
+    return reg_date
+
+
+def editor_enough_edits(global_userinfo):
+    # If, for some reason, this information hasn't come through,
+    # default to user not being valid.
+    if not global_userinfo:
+        return False
+    # Check: >= 500 edits
+    return int(global_userinfo["editcount"]) >= 500
+
+
+def editor_not_blocked(identity):
+    # If, for some reason, this information hasn't come through,
+    # default to user not being valid.
+    if not identity:
+        return False
+    # Check: not blocked
+    return identity["blocked"] == False
+
+
+def editor_account_old_enough(wp_registered):
+    # If, for some reason, this information hasn't come through,
+    # default to user not being valid.
+    if not wp_registered:
+        return False
+    # Check: registered >= 6 months ago
+    return datetime.today().date() - timedelta(days=182) >= wp_registered
+
+
+def editor_valid(username, enough_edits, account_old_enough, not_blocked):
+    """
+    Check for the eligibility criteria laid out in the terms of service.
+    Note that we won't prohibit signups or applications on this basis.
+    Coordinators have discretion to approve people who are near the cutoff.
+    """
+    if enough_edits and account_old_enough and not_blocked:
+        return True
+    else:
+        logger.info("Editor {username} was not valid.".format(username=username))
```

Comment:
Nope, just a leftover from refactor. I'm guessing this was originally dropped in for testing the validity checks.

Reply:
We're actually doing quite a bit of logging for the login process, and this happens to fall in the middle of it, here's a sample. Just noting it here for posterity.
```
Found get parameters for post-login redirection.
handshaker gotten.
Localizing oauth handshake URL.
handshaker initiated.
Authenticating user...
Identifying user...
User has been identified; getting or creating user.
Attempting to update editor after OAuth login.
Fetched global_userinfo for User.
Editor xxx was not valid.
Editor updated.
User has been updated.
```
My guess is that this was added during one of our many bouts of OAuth flakiness. I'm still going to remove this particular bit, because I agree that it doesn't seem particularly diagnostic.